src/components/Navigation/NavigationTabBar/index.tsx

```diff
@@ -61,7 +57,26 @@ function NavigationTabBar({selectedTab, isTooltipAllowed = false, isTopLevelBar
     const {orderedReports} = useSidebarOrderedReports();
     const subscriptionPlan = useSubscriptionPlan();
     const [account] = useOnyx(ONYXKEYS.ACCOUNT, {canBeMissing: false});
+    const navigationState = useNavigationState(findFocusedRoute);
+    const [lastViewedPolicy] = useOnyx(
+        ONYXKEYS.COLLECTION.POLICY,
+        {
+            canBeMissing: true,
+            selector: (val) => {
+                const {lastWorkspacesTabNavigatorRoute, workspacesTabState} = getWorkspaceNavigationRouteState();
+                const params = workspacesTabState?.routes?.at(0)?.params as WorkspaceSplitNavigatorParamList[typeof SCREENS.WORKSPACE.INITIAL];
+
+                if (!lastWorkspacesTabNavigatorRoute || lastWorkspacesTabNavigatorRoute.name !== NAVIGATORS.WORKSPACE_SPLIT_NAVIGATOR || !params?.policyID) {
+                    return undefined;
+                }
```

Comment:
Does this logic _need_ to be inside the selector? It sort of doesn't look like it does, so it's kind of a lot of logic to have run everytime the selector is ran (which is once for every policy in the collection). Can some of it be moved outside the selector?

Reply:
I think so! I like that